src/Compilers/Core/Portable/Compilation/TypeInfo.cs

```diff
@@ -35,11 +35,13 @@ public struct TypeInfo : IEquatable<TypeInfo>
         /// </summary>
         public Nullability ConvertedNullability { get; }
 
-        internal TypeInfo(ITypeSymbol type, ITypeSymbol convertedType)
+        internal TypeInfo(ITypeSymbol type, ITypeSymbol convertedType, Nullability nullability = Nullability.NotComputed, Nullability convertedNullability = Nullability.NotComputed)
```

Comment:
TypeInfo is not strongly typed.  So APIs that need to pass nullability information along now have hte unpalatable position of moving to this, but can lose the info if they were originally typed to something like an INamedTypeSymbol.

or they pass both along, but need to know that the both symbols must be the exact same instance.